ui/app/src/components/RemoteRepositoriesGrid/RemoteRepositoriesGridUI.tsx

```diff
@@ -0,0 +1,135 @@
+                  <span>
+                    <IconButton
+                      onClick={() => onClickPull(repository.name, repository.branches)}
+                      disabled={disableActions}
+                    >
+                      <ArrowDownwardRoundedIcon />
+                    </IconButton>
+                  </span>
+                </Tooltip>
+              </GlobalAppGridCell>
+              <GlobalAppGridCell>
+                <Tooltip title={<FormattedMessage id="words.pull" defaultMessage="Push" />}>
+                  <span>
+                    <IconButton
+                      onClick={() => onClickPush(repository.name, repository.branches)}
+                      disabled={disableActions}
+                    >
+                      <ArrowUpwardRoundedIcon />
+                    </IconButton>
+                  </span>
+                </Tooltip>
+              </GlobalAppGridCell>
+              <GlobalAppGridCell>
+                <ConfirmDropdown
+                  cancelText={<FormattedMessage id="words.no" defaultMessage="No" />}
+                  confirmText={<FormattedMessage id="words.yes" defaultMessage="Yes" />}
+                  confirmHelperText={
+                    <FormattedMessage id="repositories.deleteConfirmation" defaultMessage="Delete remote repository?" />
+                  }
+                  iconTooltip={<FormattedMessage id="words.delete" defaultMessage="Delete" />}
+                  icon={DeleteRoundedIcon}
+                  iconColor="action"
+                  onConfirm={() => {
+                    onDeleteRemote(repository.name);
+                  }}
+                />
+              </GlobalAppGridCell>
+            </GlobalAppGridRow>
+          ))}
+        </TableBody>
+      </Table>
+    </TableContainer>
+  );
+}
+
+export function RepositoriesGridTableHead() {
+  return (
+    <TableHead>
+      <GlobalAppGridRow className="hoverDisabled">
+        <GlobalAppGridCell className="bordered">
+          <Typography variant="subtitle2">
+            <FormattedMessage id="words.name" defaultMessage="Name" />
+          </Typography>
+        </GlobalAppGridCell>
+        <GlobalAppGridCell className="bordered">
+          <Typography variant="subtitle2">
+            <FormattedMessage id="words.url" defaultMessage="URL" />
+          </Typography>
+        </GlobalAppGridCell>
+        <GlobalAppGridCell className="bordered">
+          <Typography variant="subtitle2">
+            <FormattedMessage id="words.fetch" defaultMessage="Fetch" />
+          </Typography>
+        </GlobalAppGridCell>
+        <GlobalAppGridCell className="bordered">
+          <Typography variant="subtitle2">
+            <FormattedMessage id="repositories.pushUrl" defaultMessage="Push URL" />
+          </Typography>
+        </GlobalAppGridCell>
+        <GlobalAppGridCell className="bordered" />
+        <GlobalAppGridCell className="bordered" />
+        <GlobalAppGridCell className="bordered" />
```

Comment:
What these for?

Reply:
In the table body its for the action buttons (pull, push, delete)